src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,27 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
+                        List<PackageReference> installedProjectReferences = new List<PackageReference>(projectReferences.Count());
```

Comment:
Instantiating the list of installedProjectReferences to the number of project references would be useful to avoid array expansion if we only added the number of project references to this list but we're adding the calculatedInstalledPackages to it later. I'm not sure if the AddRange implementation considers the whole size of the new array when reallocating as to avoid multiple reallocations or not.

Reply:
@martinrrm Could you add a comment on why you think this comment should be resolved without being addressed? Looking forward to hearing your perspective! Thanks!